utils/appState.ts

```diff
@@ -0,0 +1,28 @@
+import { AppState } from "react-native";
+
+import logger from "./logger";
+
+/*
+Method to wait until the app is in Foreground (Active)
+helpful to make sure we're not doing critical database stuff
+while the database is disconnected to prevent data loss
+Will check every delayMs, then wait endDelayMs for db reconnection
+*/
+export const waitUntilAppActive = async (
```

Comment:
Approved, but I think we should avoid patterns like this especially when there are event listeners that handles this

`export const waitUntilAppActive = async (delayMs, endDelayMs, retries = 3) => {
  return new Promise((resolve, reject) => {
    let attempts = 0;

    const handleAppStateChange = (nextAppState) => {
      if (nextAppState === 'active') {
        logger.debug(`App is active, waiting ${endDelayMs}ms more`);
        setTimeout(() => {
          AppState.removeEventListener('change', handleAppStateChange);
          resolve();
        }, endDelayMs);
      } else if (attempts >= retries) {
        AppState.removeEventListener('change', handleAppStateChange);
        reject(new Error('App did not become active within the allowed retries.'));
      } else {
        attempts++;
      }
    };

    AppState.addEventListener('change', handleAppStateChange);
  });
};`

Reply:
@alexrisch apparently the PR is still unapproved?
I added a commit to use event handler, tell me if that works for you!